Snapshot-Pushback is broken

See: https://github.com/kadai-io/KadaiAdapter/actions/runs/18972498536/job/54184122750
```bash
Run ci/change_version.sh -i -m .
+ git config --global user.email ***
+ git config --global user.name ***
+ git checkout master
Switched to a new branch 'master'
M	camunda-outbox-example-boot/pom.xml
M	ci/kadai-adapter-sonar-test-coverage/pom.xml
M	kadai-adapter-camunda-8-spring-boot-example/pom.xml
M	kadai-adapter-camunda-8-system-connector/pom.xml
M	kadai-adapter-camunda-listener-example/pom.xml
M	kadai-adapter-camunda-listener/pom.xml
M	kadai-adapter-camunda-outbox-rest-spring-boot-starter/pom.xml
M	kadai-adapter-camunda-outbox-rest/pom.xml
M	kadai-adapter-camunda-spring-boot-example/pom.xml
M	kadai-adapter-camunda-spring-boot-test/pom.xml
M	kadai-adapter-camunda-system-connector/pom.xml
M	kadai-adapter-kadai-connector/pom.xml
+ git add './*pom.xml'
M	kadai-adapter-spring-boot-test/pom.xml
M	kadai-adapter/pom.xml
M	pom.xml
branch 'master' set up to track 'origin/master'.
++ increment_version 11.1.0
++ [[ ! 11.1.0 =~ [0-9]+\.[0-9]+\.[0-9]+ ]]
+++ expr 0 + 1
++ echo 11.1.1
++ increment_version 11.1.0
++ [[ ! 11.1.0 =~ [0-9]+\.[0-9]+\.[0-9]+ ]]
+++ expr 0 + 1
++ echo 11.1.1
+ git commit -m 'Updated poms to version 11.1.1-SNAPSHOT/11.1.1-SNAPSHOT'
[master 6dbb527] Updated poms to version 11.1.1-SNAPSHOT/11.1.1-SNAPSHOT
 15 files changed, 16 insertions(+), 16 deletions(-)
+ git push
remote: error: GH013: Repository rule violations found for refs/heads/master.        
remote: Review all repository rules at https://github.com/kadai-io/KadaiAdapter/rules?ref=refs%2Fheads%2Fmaster        
remote: 
remote: - Changes must be made through a pull request.        
remote: 
To https://github.com/kadai-io/KadaiAdapter
 ! [remote rejected] master -> master (push declined due to repository rule violations)
error: failed to push some refs to 'https://github.com/kadai-io/KadaiAdapter'
8s
Run ci/change_version.sh -i -m .
+ git config --global user.email ***
+ git config --global user.name ***
+ git checkout master
Switched to a new branch 'master'
M	camunda-outbox-example-boot/pom.xml
M	ci/kadai-adapter-sonar-test-coverage/pom.xml
M	kadai-adapter-camunda-8-spring-boot-example/pom.xml
M	kadai-adapter-camunda-8-system-connector/pom.xml
M	kadai-adapter-camunda-listener-example/pom.xml
M	kadai-adapter-camunda-listener/pom.xml
M	kadai-adapter-camunda-outbox-rest-spring-boot-starter/pom.xml
M	kadai-adapter-camunda-outbox-rest/pom.xml
M	kadai-adapter-camunda-spring-boot-example/pom.xml
M	kadai-adapter-camunda-spring-boot-test/pom.xml
M	kadai-adapter-camunda-system-connector/pom.xml
M	kadai-adapter-kadai-connector/pom.xml
+ git add './*pom.xml'
M	kadai-adapter-spring-boot-test/pom.xml
M	kadai-adapter/pom.xml
M	pom.xml
branch 'master' set up to track 'origin/master'.
++ increment_version 11.1.0
++ [[ ! 11.1.0 =~ [0-9]+\.[0-9]+\.[0-9]+ ]]
+++ expr 0 + 1
++ echo 11.1.1
++ increment_version 11.1.0
++ [[ ! 11.1.0 =~ [0-9]+\.[0-9]+\.[0-9]+ ]]
+++ expr 0 + 1
++ echo 11.1.1
+ git commit -m 'Updated poms to version 11.1.1-SNAPSHOT/11.1.1-SNAPSHOT'
[master 6dbb527] Updated poms to version 11.1.1-SNAPSHOT/11.1.1-SNAPSHOT
 15 files changed, 16 insertions(+), 16 deletions(-)
+ git push
remote: error: GH013: Repository rule violations found for refs/heads/master.        
remote: Review all repository rules at https://github.com/kadai-io/KadaiAdapter/rules?ref=refs%2Fheads%2Fmaster        
remote: 
remote: - Changes must be made through a pull request.        
remote: 
To https://github.com/kadai-io/KadaiAdapter
 ! [remote rejected] master -> master (push declined due to repository rule violations)
error: failed to push some refs to 'https://github.com/kadai-io/KadaiAdapter'
Error: Process completed with exit code 1.
0s

```

--- a/.github/workflows/continuous-integration.yml
+++ b/.github/workflows/continuous-integration.yml
@@ -122,6 +122,9 @@ jobs:
     runs-on: ubuntu-22.04
     name: Release artifacts to Sonatype-Central
     if: github.repository == 'kadai-io/KadaiAdapter' && ( startsWith(github.ref, 'refs/tags') || github.ref == 'refs/heads/master' ) && github.head_ref == ''
+    permissions:
+      contents: write
+      pull-requests: write
     needs: [ test ]
     # as documented in the gpg manual (https://www.gnupg.org/documentation/manuals/gnupg/Invoking-GPG_002dAGENT.html)
     # we should execute this command before interacting with gpg (otherwise gpg won't work)
@@ -200,7 +203,7 @@ jobs:
           ci/change_version.sh -i -m .
           ci/commitPoms.sh
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - name: Remove kadai artifacts from cache
         run: rm -rf ~/.m2/repository/io/kadai
       - name: Cancel workflow

--- a/.github/workflows/continuous-integration.yml
+++ b/.github/workflows/continuous-integration.yml
@@ -195,13 +195,14 @@ jobs:
           GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
           MAVEN_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
           MAVEN_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
-      - name: Update version to next snapshot and push back
+      - name: Update version to next snapshot and create PR
         run: |
           ci/change_version.sh -i -m .
           ci/commitPoms.sh
         env:
           GH_EMAIL: ${{ secrets.GH_EMAIL }}
           GH_USERNAME: ${{ secrets.GH_USERNAME }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - name: Remove kadai artifacts from cache
         run: rm -rf ~/.m2/repository/io/kadai
       - name: Cancel workflow

--- a/.github/workflows/continuous-integration.yml
+++ b/.github/workflows/continuous-integration.yml
@@ -200,8 +200,6 @@ jobs:
           ci/change_version.sh -i -m .
           ci/commitPoms.sh
         env:
-          GH_EMAIL: ${{ secrets.GH_EMAIL }}
-          GH_USERNAME: ${{ secrets.GH_USERNAME }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - name: Remove kadai artifacts from cache
         run: rm -rf ~/.m2/repository/io/kadai

--- a/ci/commitPoms.sh
+++ b/ci/commitPoms.sh
@@ -62,6 +62,7 @@ function main() {
     git checkout "$branch"
     NEXT1=$(increment_version "${BASH_REMATCH[1]}")
     NEXT2=$(increment_version "${BASH_REMATCH[2]}")
+    NEW_VERSION="${NEXT1}-SNAPSHOT/${NEXT2}-SNAPSHOT"
     PR_BRANCH="auto/version-bump-${NEXT1}-$(date +%s)"
 
     git checkout -b "$PR_BRANCH"
@@ -69,13 +70,14 @@ function main() {
     for file in "$@"; do
       [[ -n "$file" ]] && git add "$file"
     done
-    git commit -m "Updated poms to version ${NEXT1}-SNAPSHOT/${NEXT2}-SNAPSHOT"
+    git commit -m "Updated poms to version ${NEW_VERSION}"
     git push origin "$PR_BRANCH"
 
     gh pr create \
-          --title "Version bump after release ${BASH_REMATCH[1]}/${BASH_REMATCH[2]}" \
-          --body "Automated version bump after release tag ${GITHUB_REF#refs/tags/}." \
-          --base "$branch"
+          --base "$branch" \
+          --title "chore: version bump to ${NEW_VERSION}" \
+          --body "Automated version update after release. All tests have passed in the release workflow." \
+          --label "dependencies"
   else
     echo "Nothing to push - this is not a release!"
   fi

--- a/ci/commitPoms.sh
+++ b/ci/commitPoms.sh
@@ -61,12 +61,23 @@ function main() {
 
     #commit all poms
     git checkout "$branch"
+    NEXT1=$(increment_version "${BASH_REMATCH[1]}")
+    NEXT2=$(increment_version "${BASH_REMATCH[2]}")
+    PR_BRANCH="auto/version-bump-${NEXT1}-$(date +%s)"
+
+    git checkout -b "$PR_BRANCH"
     git add "./*pom.xml"
     for file in "$@"; do
       [[ -n "$file" ]] && git add "$file"
     done
-    git commit -m "Updated poms to version $(increment_version "${BASH_REMATCH[@]:1:1}")-SNAPSHOT/$(increment_version "${BASH_REMATCH[@]:2:2}")-SNAPSHOT"
-    git push
+    git commit -m "Updated poms to version ${NEXT1}-SNAPSHOT/${NEXT2}-SNAPSHOT"
+    git push origin "$PR_BRANCH"
+
+    gh pr create \
+          --title "Version bump after release ${BASH_REMATCH[1]}/${BASH_REMATCH[2]}" \
+          --body "Automated version bump after release tag ${GITHUB_REF#refs/tags/}." \
+          --base "$branch" \
+          --head "$PR_BRANCH"
   else
     echo "Nothing to push - this is not a release!"
   fi

--- a/ci/commitPoms.sh
+++ b/ci/commitPoms.sh
@@ -12,8 +12,7 @@ set -e # fail fast
 #H
 #H Requirements:
 #H   current commit is a HEAD commit
-#H   GH_USERNAME - github username / displayname (for git config)
-#H   GH_EMAIL - github email address (for git config)
+#H   GH_TOKEN - GitHub token for authentication
 #H   GITHUB_REF (format refs/tags/v[0-9]+\.[0-9]+\.[0-9]+)
 # Arguments:
 #   $1: exit code
@@ -38,7 +37,7 @@ function increment_version() {
 
 function main() {
   [[ "$1" == '-h' || "$1" == '--help' ]] && helpAndExit 0
-  [[ -z "$GH_EMAIL" || -z "$GH_USERNAME" ]] && helpAndExit 1
+  [[ -z "$GH_TOKEN" ]] && helpAndExit 1
   if [[ "$GITHUB_REF" =~ ^refs/tags/([0-9]+\.[0-9]+\.[0-9]+)/([0-9]+\.[0-9]+\.[0-9]+)$ ]]; then
     #check if tagged commit is a head commit of any branch
     commit=$(git ls-remote -q -t origin | grep "$GITHUB_REF" | cut -c1-40)
@@ -56,8 +55,8 @@ function main() {
       exit 1
     fi
     set -x
-    git config --global user.email "$GH_EMAIL"
-    git config --global user.name "$GH_USERNAME"
+    git config --global user.email "github-actions[bot]@users.noreply.github.com"
+    git config --global user.name "github-actions[bot]"
 
     #commit all poms
     git checkout "$branch"
@@ -76,8 +75,7 @@ function main() {
     gh pr create \
           --title "Version bump after release ${BASH_REMATCH[1]}/${BASH_REMATCH[2]}" \
           --body "Automated version bump after release tag ${GITHUB_REF#refs/tags/}." \
-          --base "$branch" \
-          --head "$PR_BRANCH"
+          --base "$branch"
   else
     echo "Nothing to push - this is not a release!"
   fi

